sql/plan/drop_index.go

```diff
@@ -0,0 +1,104 @@
+package plan
+
+import (
+	"github.com/sirupsen/logrus"
+	errors "gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// ErrIndexNotFound is returned when the index cannot be found.
+var ErrIndexNotFound = errors.NewKind("unable to find index %q on table %q of database %q")
+
+// DropIndex is a node to drop an index.
+type DropIndex struct {
+	Name            string
+	Table           sql.Node
+	Catalog         *sql.Catalog
+	CurrentDatabase string
+}
+
+// NewDropIndex creates a new DropIndex node.
+func NewDropIndex(name string, table sql.Node) *DropIndex {
+	return &DropIndex{name, table, nil, ""}
+}
+
+// Resolved implements the Node interface.
+func (d *DropIndex) Resolved() bool { return d.Table.Resolved() }
+
+// Schema implements the Node interface.
+func (d *DropIndex) Schema() sql.Schema { return nil }
+
+// Children implements the Node interface.
+func (d *DropIndex) Children() []sql.Node { return []sql.Node{d.Table} }
+
+// RowIter implements the Node interface.
+func (d *DropIndex) RowIter(ctx *sql.Context) (sql.RowIter, error) {
+	db, err := d.Catalog.Database(d.CurrentDatabase)
+	if err != nil {
+		return nil, err
+	}
+
+	table, ok := d.Table.(sql.Nameable)
+	if !ok {
+		return nil, ErrTableNotNameable.New()
+	}
+
+	if _, ok = db.Tables()[table.Name()]; !ok {
+		return nil, sql.ErrTableNotFound.New(table.Name())
+	}
+
+	index := d.Catalog.Index(db.Name(), d.Name)
+	if index == nil {
+		return nil, ErrIndexNotFound.New(d.Name, table.Name(), db.Name())
+	}
+	d.Catalog.ReleaseIndex(index)
+
+	done, err := d.Catalog.DeleteIndex(db.Name(), d.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	driver := d.Catalog.IndexDriver(index.Driver())
+	if driver == nil {
+		return nil, ErrInvalidIndexDriver.New(index.Driver())
+	}
+
+	go func() {
```

Comment:
should we block until the index is deleted?

Reply:
It depends. Do we want to make the user wait until it's deleted or succeed and asynchronously delete the index?

Reply:
on index deletion (that shouldn't take A LOT) I think we should wait until the index is dropped, or throw an error otherwise. WDYT @kuba-- ?

Reply:
delete index is/should be super fast (basically it deletes directory). so far it also sends delete to pilosa synchronously (because i didn't notice any lag), but If we encounter any issues we can async. call pilosa). Just to recap, it's totally fine to delete just directory. Pilosa bitmap can be deleted any time or even overwritten.

Reply:
yeah, lets block and wait if we get an error

Reply:
yeah, lets block and wait if we get an error